### PR TITLE
Aburt ModifyDateTime transformer

### PR DESCRIPTION
**Pull request by aburt-demisto**
## Status
- [ ] In Progress
- [x] Ready
- [ ] In Hold - (Reason for hold)

## Related Issues
NA

## Description
A transformer script that can take in a date / time (supported by dateparser) and apply a variation in human readable form such as "in 1 day" or "3 weeks ago". Variations that can be used are those supported by the dateparser python library.

## Screenshots
image

## Minimum version of Demisto
- [x] 4.5.0
- [ ] 5.0.0
- [ ] 5.5.0

## Does it break backward compatibility?
   - [ ] Yes
       - Further details:
   - [x] No

## Must have
- [x] Tests
- [x] Documentation 

## Demisto Partner?
NA

### CODE REVIEW

**Comment by moishce** [comment]
Note that if you enter a date with a time zone, it does not return the time zone.
Is that supposed to be this way?

**Comment by aburt-demisto** [comment]
It is. This is a simple transformer that will transform the input date by
the modification provided (such as "in 1 day" or "24 hours ago"). It is not
supposed to handle timezone transformations and it is expected that the
user enter a date / time in a zone that they expect the zone output for.

Regards

Adam


Adam Burt  |  Mgr, Solutions Architecture • EMEA

Mobile: +44(0) 7713 363155 | www.paloaltonetworks.com
<https://www.paloaltonetworks.com/>

Email: adam@paloaltonetworks.com


<https://www.paloaltonetworks.com/cortex>
<https://www.linkedin.com/company/palo-alto-networks>
<https://www.facebook.com/PaloAltoNetworks/>

The content of this message is the proprietary and confidential property of
Palo Alto Networks, and should be treated as such. If you are not the
intended recipient and have received this message in error, please delete
this message from your computer system and notify me immediately by reply
e-mail. Any unauthorized use or distribution of the content of this message
is prohibited.

File: Packs/CommonScripts/Scripts/ModifyDateTime/ModifyDateTime.py [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+from CommonServerUserPython import *  # noqa: E402 lgtm [py/polluting-import]
+import dateparser
+
+
+def apply_variation(original_datetime: datetime = None, variation: str = None) -> datetime:
+    new_time = None
+    try:
+        new_time = dateparser.parse(variation, settings={'RELATIVE_BASE': original_datetime})
+    except Exception as err:
+        return_error(f"Error adding variation to the date / time - {err}")
```
Comment (moishce) [inline Packs/CommonScripts/Scripts/ModifyDateTime/ModifyDateTime.py:12]: Screen Shot 2020-07-16 at 2 36 31
I entered an incorrect variation
and "except Exception" does not catch the error
and the "new_time" returns None

Comment (aburt-demisto) [reply Packs/CommonScripts/Scripts/ModifyDateTime/ModifyDateTime.py:12]: I have pushed new changes for checking.

**Comment by moishce** [comment]
@aburt-demisto please update your branch.
and also [You did not correct these comments for some reason or did not see them](https://github.com/demisto/content/pull/7899#discussion_r451976059)?

File: Packs/CommonScripts/Scripts/ModifyDateTime/ModifyDateTime_test.py [hunk 2]
```diff
@@ -0,0 +1,7 @@
+from ModifyDateTime import apply_variation
+args = {
+    'value': '2020/01/01',
+    'variation': 'in 1 day'
+}
+results = apply_variation(args.get('value'), args.get('variation'))
+assert results == '2020-01-02T00:00:00'
```
[the remark this answers is not in the document]
Comment (aburt-demisto) [reply Packs/CommonScripts/Scripts/ModifyDateTime/ModifyDateTime_test.py:7]: Change made

File: Packs/CommonScripts/Scripts/ModifyDateTime/ModifyDateTime.yml [hunk 3]
```diff
@@ -0,0 +1,24 @@
+args:
+- name: value
+  required: true
+  default: true
+  description: Input value
```
[the remark this answers is not in the document]
Comment (aburt-demisto) [reply Packs/CommonScripts/Scripts/ModifyDateTime/ModifyDateTime.yml]: Done.

**Comment by moishce** [comment]
@aburt-demisto The unit test failed, check why,
and check again that everything works because for me it did not work as expected

**Comment by aburt-demisto** [comment]
@moishce The unit test failed because the "main" routine takes the "value" string and converts it to a datetime object. The unit test used to allow for this, but since using only the "applyVariation" routine, it thinks the string is a datetime object, which it is not.

**Comment by aburt-demisto** [comment]
@moishce I have fixed Flake8 now

**Comment by moishce** [comment]
Looks great!
You need to update the release note by the "demisto-sdk update-release-notes"

**Comment by moishce** [comment]
Looks great!
You need to update the release note by the "demisto-sdk update-release-notes"

**Comment by aburt-demisto** [comment]
@moishce Please can you update the release notes? I have tried this with previous PRs and it's REALLY messy and complex.

**Comment by moishce** [comment]
@kirbles19 Please review the [yml file](https://github.com/demisto/content/pull/7899/files#diff-c4de428545fb22c99761df7b1600c781), the [README](https://github.com/demisto/content/pull/7899/files#diff-07e319fbea8eeb616030b6d43a41b7c8) and the [release note](https://github.com/demisto/content/pull/7899/files#diff-6b3c517b67dca05c6841610c377b0da4)

**Comment by kirbles19** [comment]
@aburt-demisto Done

**Pull request merged**